// @nextui-org/react/esm/checkbox/checkbox-group.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

exports.__esModule = true;
exports.default = void 0;

var _react = _interopRequireWildcard(require("react"));

var _checkboxContext = require("./checkbox-context");

var _useWarning = _interopRequireDefault(require("../use-warning"));

var _checkbox = require("./checkbox.styles");

var _withDefaults = _interopRequireDefault(require("../utils/with-defaults"));

var _assertion = require("../utils/assertion");

var _jsxRuntime = require("react/jsx-runtime");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function (nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || typeof obj !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

const defaultProps = {
  color: 'default',
  labelColor: 'default',
  disabled: false,
  size: 'md'
};

const CheckboxGroup = ({
  color,
  labelColor,
  disabled,
  onChange,
  value,
  size,
  children,
  className,
  style,
  ...props
}) => {
  const [selfVal, setSelfVal] = (0, _react.useState)([]);

  if (!value && _assertion.__DEV__) {
    value = [];
    (0, _useWarning.default)('Props "value" is required.', 'Checkbox Group');
  }

  const updateState = (val, checked) => {
    const removed = selfVal.filter(v => v !== val);
    const next = checked ? [...removed, val] : removed;
    setSelfVal(next);
    onChange && onChange(next);
  };

  const providerValue = (0, _react.useMemo)(() => {
    return {
      updateState,
      color,
      labelColor,
      size,
      disabledAll: disabled,
      inGroup: true,
      values: selfVal
    };
  }, [disabled, selfVal]);
  (0, _react.useEffect)(() => {
    setSelfVal(value);
  }, [value.join(',')]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_checkboxContext.CheckboxContext.Provider, {
    value: providerValue,
    children: /*#__PURE__*/(0, _jsxRuntime.jsx)(_checkbox.StyledCheckboxGroup, {
      role: "group",
      size: size,
      ...props,
      children: children
    })
  });
};

CheckboxGroup.toString = () => '.nextui-checkbox-group';

var _default = (0, _withDefaults.default)(CheckboxGroup, defaultProps);

exports.default = _default;
module.exports = exports.default;
