// @nextui-org/react/esm/image/image.styles.js
"use strict";

exports.__esModule = true;
exports.StyledImageSkeleton = exports.StyledImageContainer = exports.StyledImage = void 0;

var _stitches = require("../theme/stitches.config");

const loading = (0, _stitches.keyframes)({
  '0%': {
    backgroundPosition: '200% 0'
  },
  to: {
    backgroundPosition: '-200% 0'
  }
});
const StyledImageContainer = (0, _stitches.styled)('div', {
  opacity: 0,
  margin: '0 auto',
  position: 'relative',
  br: '$lg',
  overflow: 'hidden',
  maxWidth: '100%',
  transition: 'transform 250ms ease 0ms, opacity 200ms ease-in 0ms',
  '@motion': {
    transition: 'none'
  },
  variants: {
    ready: {
      true: {
        opacity: 1
      },
      false: {
        opacity: 0
      }
    }
  }
});
exports.StyledImageContainer = StyledImageContainer;
const StyledImage = (0, _stitches.styled)('img', {
  size: '100%',
  display: 'block'
});
exports.StyledImage = StyledImage;
const StyledImageSkeleton = (0, _stitches.styled)('div', {
  position: 'absolute',
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  size: '100%',
  borderRadius: '$lg',
  backgroundImage: 'linear-gradient(270deg,$accents1,$accents2,$accents2,$accents1)',
  backgroundSize: '400% 100%',
  animation: `${loading} 5s ease-in-out infinite`,
  transition: 'opacity 300ms ease-out'
});
exports.StyledImageSkeleton = StyledImageSkeleton;
