// @nextui-org/react/cjs/radio/radio.styles.js
"use strict";

exports.__esModule = true;
exports.StyledRadioPoint = exports.StyledRadioLabel = exports.StyledRadioInput = exports.StyledRadioGroup = exports.StyledRadioDescription = exports.StyledRadio = void 0;

var _stitches = require("../theme/stitches.config");

const StyledRadioInput = (0, _stitches.styled)('input', {}, _stitches.sharedVisuallyHidden);
exports.StyledRadioInput = StyledRadioInput;
const StyledRadioLabel = (0, _stitches.styled)('span', {
  fontSize: '$$radioSize',
  us: 'none',
  d: 'inline-flex',
  ai: 'center',
  variants: {
    color: {
      default: {
        color: '$text'
      },
      primary: {
        color: '$primary'
      },
      secondary: {
        color: '$secondary'
      },
      success: {
        color: '$success'
      },
      warning: {
        color: '$warning'
      },
      error: {
        color: '$error'
      }
    },
    disabled: {
      true: {
        color: '$accents4'
      }
    }
  }
});
exports.StyledRadioLabel = StyledRadioLabel;
const StyledRadioPoint = (0, _stitches.styled)('span', {
  size: '$$radioSize',
  br: '$$radioRadii',
  position: 'relative',
  d: 'inline-block',
  mr: 'calc($$radioSize * 0.375)',
  '&:after': {
    content: '',
    d: 'block',
    position: 'absolute',
    size: '$$radioSize',
    br: '$$radioRadii',
    boxSizing: 'border-box',
    border: '2px solid $border'
  }
}, _stitches.sharedFocus);
exports.StyledRadioPoint = StyledRadioPoint;
const StyledRadio = (0, _stitches.styled)('label', {
  d: 'flex',
  w: 'initial',
  ai: 'flex-start',
  position: 'relative',
  fd: 'column',
  jc: 'flex-start',
  cursor: 'pointer',
  '@motion': {
    [`& ${StyledRadioPoint}`]: {
      transition: 'none',
      '&:after': {
        transition: 'none'
      }
    }
  },
  variants: {
    color: {
      default: {
        $$radioColor: '$colors$primary',
        $$radioColorHover: '$colors$primaryDark'
      },
      primary: {
        $$radioColor: '$colors$primary',
        $$radioColorHover: '$colors$primaryDark'
      },
      secondary: {
        $$radioColor: '$colors$secondary',
        $$radioColorHover: '$colors$secondaryDark'
      },
      success: {
        $$radioColor: '$colors$success',
        $$radioColorHover: '$colors$successDark'
      },
      warning: {
        $$radioColor: '$colors$warning',
        $$radioColorHover: '$colors$warningDark'
      },
      error: {
        $$radioColor: '$colors$error',
        $$radioColorHover: '$colors$errorDark'
      }
    },
    size: {
      xs: {
        $$radioSize: '$space$7'
      },
      sm: {
        $$radioSize: '$space$8'
      },
      md: {
        $$radioSize: '$space$9'
      },
      lg: {
        $$radioSize: '$space$10'
      },
      xl: {
        $$radioSize: '$space$11'
      }
    },
    disabled: {
      true: {
        cursor: 'not-allowed',
        $$radioColor: '$colors$accents4'
      }
    },
    squared: {
      true: {
        $$radioRadii: '$space$1'
      },
      false: {
        $$radioRadii: '$radii$rounded'
      }
    },
    active: {
      true: {
        [`& ${StyledRadioPoint}`]: {
          '&:after': {
            border: 'calc($$radioSize * 0.34) solid $$radioColor'
          }
        },
        '&:hover': {
          [`& ${StyledRadioPoint}`]: {
            '&:after': {
              border: 'calc($$radioSize * 0.34) solid $$radioColorHover'
            }
          }
        }
      }
    },
    animated: {
      true: {
        [`& ${StyledRadioPoint}`]: {
          transition: '$default',
          '&:after': {
            transition: '$default'
          }
        }
      },
      false: {
        [`& ${StyledRadioPoint}`]: {
          transition: 'none',
          '&:after': {
            transition: 'none'
          }
        }
      }
    }
  },
  defaultVariants: {
    size: 'md',
    squared: false,
    animated: true
  },
  compoundVariants: [{
    // !active && !disabled
    active: false,
    disabled: false,
    css: {
      '&:hover': {
        [`& ${StyledRadioPoint}`]: {
          bg: '$border'
        }
      }
    }
  }]
});
exports.StyledRadio = StyledRadio;
const StyledRadioGroup = (0, _stitches.styled)('div', {
  border: 0,
  margin: 0,
  padding: 0,
  display: 'flex',
  fd: 'column',
  variants: {
    size: {
      xs: {
        $$radioGroupGap: '$space$7'
      },
      sm: {
        $$radioGroupGap: '$space$8'
      },
      md: {
        $$radioGroupGap: '$space$9'
      },
      lg: {
        $$radioGroupGap: '$space$10'
      },
      xl: {
        $$radioGroupGap: '$space$11'
      }
    },
    row: {
      true: {
        fd: 'row',
        [`& ${StyledRadio}`]: {
          marginTop: 0,
          marginRight: '$$radioGroupGap'
        }
      },
      false: {
        [`& ${StyledRadio}`]: {
          marginTop: '$$radioGroupGap',
          marginRight: 0
        }
      }
    }
  },
  defaultVariants: {
    row: false
  }
});
exports.StyledRadioGroup = StyledRadioGroup;
const StyledRadioDescription = (0, _stitches.styled)('span', {
  color: '$accents4',
  fontSize: 'calc($$radioSize * 0.85)',
  paddingLeft: 'calc($$radioSize + $$radioSize * 0.375)'
}); // types

exports.StyledRadioDescription = StyledRadioDescription;
