// @nextui-org/react/esm/user/user.styles.js
"use strict";

exports.__esModule = true;
exports.StyledUserSocial = exports.StyledUserName = exports.StyledUserLink = exports.StyledUserInfo = exports.StyledUser = void 0;

var _stitches = require("../theme/stitches.config");

const StyledUser = (0, _stitches.styled)('div', {
  d: 'inline-flex',
  p: '0 $sm',
  jc: 'center',
  ai: 'center',
  w: 'max-content',
  maxWidth: '100%'
});
exports.StyledUser = StyledUser;
const StyledUserInfo = (0, _stitches.styled)('div', {
  ml: '$sm',
  d: 'inline-flex',
  fd: 'column',
  whiteSpace: 'nowrap'
});
exports.StyledUserInfo = StyledUserInfo;
const StyledUserName = (0, _stitches.styled)('span', {
  fontSize: '$xs',
  color: '$text',
  lh: '$sm',
  tt: 'capitalize',
  fontWeight: '$medium',
  maxW: '$60',
  to: 'ellipsis',
  // text overflow
  ov: 'hidden' // overflow

});
exports.StyledUserName = StyledUserName;
const StyledUserSocial = (0, _stitches.styled)('span', {
  fontSize: '$tiny',
  color: '$accents4',
  '*:first-child': {
    mt: 0
  },
  '*:last-child': {
    mb: 0
  }
});
exports.StyledUserSocial = StyledUserSocial;
const StyledUserLink = (0, _stitches.styled)('a', {
  a: {
    '&:hover': {
      opacity: 0.7
    }
  }
}); // types

exports.StyledUserLink = StyledUserLink;
